### DISCUSSION TITLE
Random field value

### DISCUSSION THREAD

**Discussion by uprog89** [directus/directus#2707] [Archive]
How to create a field where data under type AD0101230ZD will be generated in it when creating a record?

I tried through js, I could not find the field and fill it with data.

**Comment by rijkvanzanten** [comment]
I think your best bet is to create a filter hook and set the value of a particular field according to your specifications.

**Reply by uprog89** [reply]
Help how to do this?

**Reply by rijkvanzanten** [reply]
See https://docs.directus.io/extensions/hooks.html#hooks

**Reply by uprog89** [reply]
Created a file in a directory. How to connect it? Or how to check the operation or enable debug

**Reply by uprog89** [reply]
Understood more or less! Can you show an example when creating an entry in the add text field?

**Comment by uprog89** [comment]
hook made. It works after saving. How to make it work when creating visually?
On the creation page

**Reply by rijkvanzanten** [reply]
You could use a readonly text-input to view it. If you want to generate the value client side, you'll have to create a custom interface (https://docs.directus.io/extensions/interfaces.html#files-structure)
